src/commands/whoami/mod.rs

```diff
@@ -1,20 +1,56 @@
+use crate::http;
 use crate::settings::global_user::GlobalUser;
-use crate::terminal::{emoji, message};
+use crate::terminal::emoji;
+
+use cloudflare::endpoints::account;
+use cloudflare::framework::apiclient::ApiClient;
+use cloudflare::framework::HttpApiClientConfig;
+
+use prettytable::{Cell, Row, Table};
 
 pub fn whoami(user: &GlobalUser) -> Result<(), failure::Error> {
     // If using email + API key for auth, simply prints out email from config file.
-    let email: String = match user {
-        GlobalUser::GlobalKeyAuth { email, .. } => email.to_string(),
-        GlobalUser::TokenAuth { .. } => failure::bail!(
-            "At the moment, Wrangler cannot get user information for users using API tokens"
-        ),
+    let table: String = match user {
+        GlobalUser::GlobalKeyAuth { email, .. } => {
+            println!(
+                "\n{} You are logged in with a Global API Key, associated with the email '{}'.\n",
+                emoji::WAVING,
+                email,
+            );
+
+            fetch_accounts(user)?
+        }
+        GlobalUser::TokenAuth { .. } => {
+            println!("\n{} You are logged in with an API Token.\n", emoji::WAVING,);
+
+            fetch_accounts(user)?
+        }
     };
 
-    let msg = format!(
-        "{} You are logged in with the email '{}'.",
-        emoji::WAVING,
-        email
-    );
-    message::info(&msg);
+    println!("{}", &table);
     Ok(())
 }
+
+fn fetch_accounts(user: &GlobalUser) -> Result<String, failure::Error> {
+    let client = http::cf_v4_api_client(user, HttpApiClientConfig::default())?;
+    let response = client.request(&account::ListAccounts { params: None })?;
+
+    let mut table = Table::new();
+    let table_head = Row::new(vec![Cell::new("Account Name"), Cell::new("Account ID")]);
+    table.add_row(table_head);
+
+    match user {
+        GlobalUser::TokenAuth { .. } => {
```

Comment:
nit: can we use `if let` syntax here instead of `match`

```rust
if let GlobalUser::TokenAuth { .. } = user {
```

Reply:
i think the match is more appropriate, fwiw.

Reply:
it's only matching on one thing, i think generally `if let` fits that model better, it doesn't need an `else` and would eliminate the `_ => ()`

Reply:
can you say more about why you want if let?

Reply:
(there's no assignment here so i am less inclined)